### PR TITLE
feat: Added bn_BD Bangla language

### PR DESCRIPTION
**Pull request by AminulBD**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
Re-Opened because this #31252 pull has the wrong branch.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Added bn_BD (Bangla) locale. |
| 🇨🇳 Chinese | 添加了 bn_BD（孟加拉语）语言环境。|

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
Missing `/locale-provider/bn_BD.tsx` file.

**Comment by yoyo837** [comment]
Add test case in `/locale-provider/__tests__/index.test.js` file for `bn_BD`.

File: components/locale/bn_BD.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import Pagination from 'rc-pagination/lib/locale/bn_BD';
+import DatePicker from '../date-picker/locale/bn_BD';
+import TimePicker from '../time-picker/locale/bn_BD';
+import Calendar from '../calendar/locale/bn_BD';
+import { Locale } from '../locale-provider';
+
+const typeTemplate = '${label} টি সঠিক ${type} নয়।';
+
+const localeValues: Locale = {
+  locale: 'bn',
```
Comment (yoyo837) [inline components/locale/bn_BD.tsx:11]: image

Comment (yoyo837) [reply components/locale/bn_BD.tsx:11]: Which one is your right choice?

Comment (AminulBD) [reply components/locale/bn_BD.tsx:11]: `bn-bd` is correct. But, `bn` and `bn-bd` use the same strings and there is no difference. Should I language for both?

Comment (yoyo837) [reply components/locale/bn_BD.tsx:11]: No, just only switch to `bn-bd`.

Comment (AminulBD) [reply components/locale/bn_BD.tsx]: switched.

**Comment by yoyo837** [comment]
Please update the snapshot.

**Review by yoyo837 — APPROVED** [review]

**Comment by yoyo837** [comment]
i18n Docs update needed.

**Pull request merged**